### PR TITLE
Drop `audit.k8s.io/{v1alpha1/v1beta1}`

### PR DESCRIPTION
**Pull request by acumino**
**How to categorize this PR?**

/area open-source
/kind cleanup

**What this PR does / why we need it**:
https://github.com/gardener/gardener/pull/6101 PR added `audit.k8s.io/{v1alpha1/v1beta1}` to support older version of Kubernetes in gardener. With https://github.com/gardener/gardener/pull/8487 support for Kubernetes veriosn <1.24 has been dropped, 1.23 was the highest Kubernetes version which supported `audit.k8s.io/v1beta1`. With end of support of v1.23 we no longer need to have `audit.k8s.io/{v1alpha1/v1beta1}`.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:
/cc @shafeeqes 

**Release note**:

```other operator
NONE
```

### CODE REVIEW

File: pkg/admissioncontroller/webhook/admission/auditpolicy/handler_test.go [hunk 1]
```diff
@@ -369,64 +341,6 @@ rules:
...
-					TypeMeta:   metav1.TypeMeta{},
-					ObjectMeta: metav1.ObjectMeta{ResourceVersion: "2"},
-					Data:       map[string]string{"policy": validAuditPolicyV1alpha1},
-				}
-				mockReader.EXPECT().Get(gomock.Any(), kubernetesutils.Key(shootNamespace, cmName), gomock.AssignableToTypeOf(&corev1.ConfigMap{})).DoAndReturn(func(_ context.Context, key client.ObjectKey, cm *corev1.ConfigMap, _ ...client.GetOption) error {
-					*cm = returnedCm
-					return nil
-				})
-				test(admissionv1.Create, nil, shootv1beta1, false, statusCodeInvalid, "audit policy with apiVersion 'v1alpha1' is not supported", "")
-			})
```
Comment (shafeeqes) [inline pkg/admissioncontroller/webhook/admission/auditpolicy/handler_test.go]: Can you keep one test and expect an error message which says failing to decode now that we removed it from the scheme?

Comment (acumino) [reply pkg/admissioncontroller/webhook/admission/auditpolicy/handler_test.go]: Added test in ba5abe5. PTAL

**Review by shafeeqes — COMMENTED** [review]
Thanks!
/lgtm

**Comment by vpnachev** [comment]
Conceptually wise
/lgtm

Code wise, somebody else has to take a look.

**Pull request merged**